src/EnergyPlus/CurveManager.cc

```diff
@@ -6379,6 +6419,508 @@ Label999: ;
+		std::string & FileName,
+		bool IndVarSwitch,
+		int & lineNum
+	)
+	{
+
+	// SUBROUTINE INFORMATION:
+	//       AUTHOR         Lixing Gu, FSEC
+	//       DATE WRITTEN   Feb. 2017
+	//       MODIFIED       na
+	//       RE-ENGINEERED  na
+
+	// PURPOSE OF THIS FUNCTION:
+	// get data from an external file used to retrieve optical properties
+
+	// METHODOLOGY EMPLOYED:
+	// Recommended by NREL to read data
+
+	// Using/Aliasing
+	using DataSystemVariables::CheckForActualFileName;
+	using DataSystemVariables::TempFullFileName;
+	using General::splitString;
+
+	int TableNum;
+	bool FileExists;
+	std::ifstream infile( FileName );
+	std::string line;
+
+	CheckForActualFileName( FileName, FileExists, TempFullFileName );
+	if ( !FileExists ) {
+		ShowSevereError( "CurveManager: SearchTableDataFile: Could not open Table Data File, expecting it as file name = " + FileName );
+		ShowContinueError( "Certain run environments require a full path to be included with the file name in the input field." );
+		ShowContinueError( "Try again with putting full path and file name in the field." );
+		ShowFatalError( "Program terminates due to these conditions." );
+	}
+
+	TableNum = PerfCurve( CurveNum ).TableIndex;
+	TableData( TableNum ).X1.allocate( 0 );
+	TableData( TableNum ).X2.allocate( 0 );
+	TableData( TableNum ).Y.allocate( 0 );
+	lineNum = 0;
+	while (std::getline( infile, line )) {
+		std::vector<std::string> strings = splitString( line, ',' );
+		lineNum++;
+		if (strings.size() > 2) {
+			if (IndVarSwitch) {
+				TableData( TableNum ).X1.push_back( std::stod( strings[1] ) );
+				TableData( TableNum ).X2.push_back( std::stod( strings[0] ) );
+			} else {
+				TableData( TableNum ).X1.push_back( std::stod( strings[0] ) );
+				TableData( TableNum ).X2.push_back( std::stod( strings[1] ) );
+			}
+			TableData( TableNum ).Y.push_back( std::stod( strings[2] ) );
+		}
+	}
+
+	}
+
+	void
+	SetSameIndeVariableValues(
+		int const TransCurveIndex,
+		int const FRefleCurveIndex,
+		int const BRefleCurveIndex
+	)
+	{
+
+		// SUBROUTINE INFORMATION:
+		//       AUTHOR         Lixing Gu, FSEC
+		//       DATE WRITTEN   Feb. 2017
+		//       MODIFIED       na
+		//       RE-ENGINEERED  na
+
+		// PURPOSE OF THIS FUNCTION:
+		// Set up indenpendent varilable values the same in 3 table data 
+
+		// METHODOLOGY EMPLOYED:
+		// Recommended by NREL to read data
+
+		// Using/Aliasing
+		int X1TableNum;
+		int X2TableNum;
+		Array1D< Real64 > XX1;
+		Array1D< Real64 > XX2;
+		int i;
+		int j;
+		int k;
+		int l;
+		int m;
+		Real64 XX;
+		Real64 X1;
+		Real64 X2;
+		Real64 YY;
+		Array1D< int > Tables;
+		int TableNum;
+
+		Tables.allocate( 3 );
+		const Real64 tol = 1.0e-5;
+		bool found;
+
+		Tables( 1 ) = PerfCurve( TransCurveIndex ).TableIndex;
+		Tables( 2 ) = PerfCurve( FRefleCurveIndex ).TableIndex;
+		Tables( 3 ) = PerfCurve( BRefleCurveIndex ).TableIndex;
+
+		// Set up independent variable size to cover all independent varaible values in 3 tables
```

Comment:
Thanks. I am going to revise them.

Actually, I found many typos in other modules. I will also fix them. See a screen shot below.

![capture](https://cloud.githubusercontent.com/assets/5970656/25912160/516445b2-3584-11e7-9962-4975d02cc14c.JPG)